### PR TITLE
Bug 1289574 - Handle various edge cases with bugzilla mirroring

### PR DESCRIPTION
**Pull request by wlach**
* UTF-8 characters in summary
* Summary exceeds 255 characters

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1732)

### CODE REVIEW

File: treeherder/etl/bugzilla.py [hunk 1]
```diff
@@ -45,15 +51,19 @@ def run(self):
...
+                            'resolution': bug.get('resolution', ''),
+                            'summary': smart_text(
+                                bug.get('summary', '')[:max_summary_length]),
+                            'crash_signature': bug.get('cf_crash_signature', ''),
+                            'keywords': ",".join(bug['keywords']),
+                            'os': bug.get('op_sys', ''),
+                            'modified': dateutil.parser.parse(
+                                bug['last_change_time'], ignoretz=True)
+                        })
+                except Exception, e:
```
Comment (edmorley) [inline treeherder/etl/bugzilla.py]: This is Python <=2.5 syntax, Python 2.6 support (and recommend) `Exception as e` which is Python 3 compatible :-)

https://www.python.org/dev/peps/pep-3110/#compatibility

**Comment by edmorley** [comment]
Thank you for fixing this! :-)

**Comment by jgraham** [comment]
Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/treeherder/1732)*

**Comment by jgraham** [comment]
Seems fine except you broke the tests; the requirement for the caller to provide the normalisation seems suboptimal in light of this. Maybe consider writing a [custom field type](https://docs.djangoproject.com/en/1.9/howto/custom-model-fields/)? It doesn't seem like Django has a simpler method than this :(

**Comment by wlach** [comment]
Writing a custom field type seems like overkill to me. Let's just duplicate the business logic to handle utf-8 / long summary fields inside the unit test.

**Comment by wlach** [comment]
Ok, made the unit tests pass. @jgraham let me know if my naive approach is ok.

**Comment by jgraham** [comment]
Reviewed 4 of 4 files at r4, 1 of 1 files at r5, 4 of 4 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/treeherder/1732)*

**Comment by jgraham** [comment]
Yeah, I guess this will do for now, although I don't think we want to violate DRY with data normalisation code like this (I'm sure I have also done so at times, but I think there is a real maintenance burden associated with this approach).

**Pull request merged**